Do not hardcode -ldl library name (dlopen() is in libc on NetBSD)

--- src/CMakeLists.txt~	2011-06-21 19:44:23.000000000 +0200
+++ src/CMakeLists.txt	2011-10-29 13:12:23.000000000 +0200
@@ -53,7 +53,7 @@
 PKG_CONFIG_USE_DEPENDENCY(${LIBRARY_NAME} jrl-mal)
 
 IF (UNIX)
-  TARGET_LINK_LIBRARIES(${LIBRARY_NAME} dl pthread)
+  TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${CMAKE_DL_LIBS} pthread)
 ENDIF (UNIX)
 
 TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${Boost_LIBRARIES})
